theano/scan_module/scan_opt.py

```diff
@@ -565,20 +630,18 @@ def add_to_replace(y):
         clean_replace_with_in = []
         clean_replace_with_out = []
 
-        existent_nodes = [nd for nd in local_fgraph_topo
-                          if nd not in to_remove_set]
+        existent_nodes = filter(lambda n: n not in to_remove_set, local_fgraph_topo)
```

Comment:
Here, remove local_fgraph_topo from this opt, it isn't needed. Use local_fgraph.apply_nodes here.